pandas/io/sql.py

```diff
@@ -1698,10 +1941,21 @@ def to_sql(
         frame : DataFrame
         name : string
             Name of SQL table.
-        if_exists : {'fail', 'replace', 'append'}, default 'fail'
+        if_exists : {'fail', 'replace', 'append'},
+            default 'fail'.
             - fail: If table exists, do nothing.
-            - replace: If table exists, drop it, recreate it, and insert data.
+            - replace: If table exRsts, drop it, recreate it, and insert data.
             - append: If table exists, insert data. Create if does not exist.
+        on_conflict : {None, 'do_nothing', 'do_update'}, optional
+            Determine insertion behaviour in case of a primary key clash.
+            If the table being written has primary key constraints, attempting
+            to insert new rows with the same values in the primary key columns,
+            will cause an error.  In this case the conflicting records can either
```

Comment:
`throw an Exception` opposed to cause an error. And name the type of exception

Reply:
If a user attempts to append to a pkey-constrained table when `on_conflict` is `None`, and there are pkey clashes, it raises a rather descriptive error around the failed `uniqueness constraint` on the table.  I find that to be a pretty useful error message.  Is that something we can just propagate, or does the exception need to be caught and raised as a custom exception from within pandas?

Reply:
it seems like this exception would always be the same so clarifying in the documentation about the type of the exception that sqlalchemy throws would be helpful (eg. `sqlalchemy.UniqueConstraint` if that was the exception type)